internal/reconciliations/istio/configuration.go

```diff
@@ -86,6 +113,34 @@ func CheckIstioVersion(currentIstioVersionString, targetIstioVersionString strin
 	return nil
 }
 
+func IngressGatewayNeedsRestart(istioCR operatorv1alpha1.Istio) (bool, error) {
```

Comment:
Do we want/need to expose this?

Reply:
I did refactor it so its a single method now, but unfortunately we need to export it because it is used by the tests that are in `xxx_test` package.

Reply:
did refactor test file to use same package as the implementation, as we discussed.